pkg/render/prometheus_api.go

```diff
@@ -0,0 +1,303 @@
+)
+
+const (
+	TigeraPrometheusAPIName = "tigera-prometheus-api"
+
+	calicoNodePrometheusServiceName = "calico-node-prometheus"
+
+	prometheusEndpointUrlEnvVarName     = "PROMETHEUS_ENDPOINT_URL"
+	prometheusOperatedHttpServiceScheme = "http"
+	prometheusOperatedHttpServiceHost   = PrometheusHTTPAPIServiceName + ".tigera-prometheus"
+
+	prometheusServiceListenAddrEnvVarName = "LISTEN_ADDR"
+
+	tigeraPrometheusServiceHealthEndpoint = "/health"
+
+	// pod listens to and the TargetPort of calico-node-prometheus service.
+	// If not set it defaults to port 9090. In the scenario that the cluster
+	// is hosted on EKS and using Calico as it's CNI, tigera-prometheus-service
+	// will be HostNetwoked. This is configurable in the configMap passed to TigeraPrometheusAPI
+	tigeraPrometheusAPIListenPortFieldName = "tigeraPrometheusAPIListenPort"
+)
+
+func TigeraPrometheusAPI(cr *operator.InstallationSpec, pullSecrets []*corev1.Secret, configMap *corev1.ConfigMap) (Component, error) {
+
+	return &tigeraPrometheusAPIComponent{
+		configMap:    configMap,
+		installation: cr,
+		pullSecrets:  pullSecrets,
+	}, nil
+}
+
+type tigeraPrometheusAPIComponent struct {
+	configMap              *corev1.ConfigMap
+	installation           *operator.InstallationSpec
+	pullSecrets            []*corev1.Secret
+	prometheusServiceImage string
+}
+
+func (p *tigeraPrometheusAPIComponent) ResolveImages(is *operator.ImageSet) error {
+	reg := p.installation.Registry
+	path := p.installation.ImagePath
+	prefix := p.installation.ImagePrefix
+
+	prometheusServiceImage, err := components.GetReference(components.ComponentTigeraPrometheusService, reg, path, prefix, is)
+	if err != nil {
+		return err
+	}
+
+	p.prometheusServiceImage = prometheusServiceImage
+
+	return nil
+}
+
+// Objects returns the lists of objects in this component that should be created and/or deleted during
+// rendering.
+func (p *tigeraPrometheusAPIComponent) Objects() (objsToCreate, objsToDelete []client.Object) {
+	// tigera-prometheus-objects
+	namespacedObjects := []client.Object{}
+
+	// place pullsecrets secrets under tigera-prometheus
+	secrets := secret.CopyToNamespace(common.TigeraPrometheusNamespace, p.pullSecrets...)
+	namespacedObjects = append(namespacedObjects, secret.ToRuntimeObjects(secrets...)...)
+
+	tigeraPrometheusApiConfigMap := p.configMap
+
+	if tigeraPrometheusApiConfigMap == nil {
+		tigeraPrometheusApiConfigMap = p.getDefaultConfigMap()
+		namespacedObjects = append(namespacedObjects, tigeraPrometheusApiConfigMap)
+	}
+
+	configuredTigeraPrometheusApiPort, err := strconv.Atoi(tigeraPrometheusApiConfigMap.Data[tigeraPrometheusAPIListenPortFieldName])
+
+	if err != nil {
+		log.Error(err, fmt.Sprintf("incorrect listen port value for %s, defaulting to port: %d", TigeraPrometheusAPIName, PrometheusDefaultPort))
+		configuredTigeraPrometheusApiPort = PrometheusDefaultPort
+	}
+
+	namespacedObjects = append(
+		namespacedObjects,
+		p.podSecurityPolicy(),
```

Comment:
I think adding a PSP might need to be dependent on not being on openshift, at least from looking at other renders.
Have you tried this on an openshift install?

Reply:
thanks for catching that, it didnt end up deploying the images with PSPs on in openshift 

https://github.com/tigera/operator/pull/1379/commits/9f54439f513d948e141eaec7de4808449f8d3b53